rfcs/0000-rfc-process.md

```diff
@@ -0,0 +1,148 @@
+and controlled path for new features to enter the nix language, packages and
+OS, so that all stakeholders can be confident about the direction the
+ecosystem is evolving in.
+
+This RFC is used to bootstrap the process. Further RFCs can be used to refine
+the process.
+
+# Motivation
+[motivation]: #motivation
+
+There are a number of changes that are significant enough that they could
+benefit from wider community consensus before being implemented. Either
+because they introduce new concepts, big changes or are controversial enough
+that not everybody will agree on the direction to take.
+
+# Detailed design
+[design]: #detailed-design
+
+Many changes, including bug fixes and documentation improvements can be
+implemented and reviewed via the normal GitHub pull request workflow.
+
+Some changes though are "substantial", and we ask that these be put through a
+bit of a design process and produce a consensus among the Nix community.
+
+This is the bulk of the RFC. Explain the design in enough detail for somebody
+familiar with the ecosystem to understand, and implement.  This should get
+into specifics and corner-cases, and include examples of how the feature is
+used.
+
+## When you need to follow this process
+
+You need to follow this process if you intend to make "substantial" changes to
+the Nix ecosystem. What constitutes a "substantial" change is evolving based
+on community norms, but may include the following.
+
+* Any semantic or syntactic change to the language that is not a bugfix
+* Removing language features
+* Big restructuring of nixpkgs
+* Expansions to the scope of nixpkgs (new arch, major subprojects, ...)
+* Introduction of new interfaces or functions
+* A controversial change
+
+Some changes do not require an RFC:
+
+* Adding, updating and removing packages in nixpkgs
+* Fixing security updates and bugs that don't break interfaces
+
+If you submit a pull request to implement a new feature without going
+through the RFC process, it may be closed with a polite request to
+submit an RFC first.
+
+## What the process is
+
+In short, to get a major feature added to the Nix ecosystem, one should first
+go throught the RFC process in order to improve the likelyhood of inclusion.
+Here are roughly the steps that one would take:
+
+* Fork the RFC repo https://github.com/NixOS/rfcs
+* Copy `0000-template.md` to `rfcs/0000-my-feature.md` (where 'my-feature' is
+  descriptive. don't assign an RFC number yet).
+* Fill in the RFC
+* Submit a pull request. Rename the rfcs with the PR number. (eg: PR #123 would
+  be `rfcs/0123-my-feature.md`)
+* Get initial feedback on the PR.
+
+At this point, the person submitting the RFC should find at least one "co-author"
+that will help them bring the RFC to completion. The goal is to improve the
+chances that the RFC is both desired and likely to be implemented.
+
+Once the author is happy with the state of the RFC, he tells it's ready for
```

Comment:
@grahamc Agreed, any idea on how to improve the sentence?

@FRidh Mailing list seems reasonable. The process is a bit vague but it's intentional, I would prefer to update the process based on current usage rather than come up with something too artificial. Rust has multiple sub-teams each responsible for their RFC process but we don't have that. Potentially we could start forming sub-teams in a further RFC if we feel like it would be useful.